### PR TITLE
Support enabling opentracing by user

### PR DESCRIPTION
**Pull request by richvdh**
Add a config option which allows enabling opentracing by user id, eg for
debugging requests made by a test user.

### CODE REVIEW

**Comment by richvdh** [comment]
this is inspired by my hunt for delayed to-device messages (#9564)

File: synapse/api/auth.py [hunk 1]
```diff
@@ -260,6 +264,13 @@ async def get_user_by_req(
             opentracing.set_tag("user_id", user_info.user_id)
             if device_id:
                 opentracing.set_tag("device_id", device_id)
+            if user_info.token_owner in self._force_tracing_for_users:
+                logger.info("enabling tracing for %s", user_info.token_owner)
+                opentracing.set_tag(opentracing.tags.SAMPLING_PRIORITY, 1)
+            else:
+                logger.info(
+                    "%s not in %s", user_info.token_owner, self._force_tracing_for_users
+                )
```
Comment (erikjohnston) [inline synapse/api/auth.py:273]: Do we want this at info?

Comment (richvdh) [reply synapse/api/auth.py:273]: er no. Sorry, I left the debugging in.

File: synapse/api/auth.py [hunk 2]
```diff
@@ -88,6 +88,8 @@ def __init__(self, hs: "HomeServer"):
         self._track_appservice_user_ips = hs.config.track_appservice_user_ips
         self._macaroon_secret_key = hs.config.macaroon_secret_key
```
Comment (richvdh) [inline synapse/api/auth.py:90]:
```suggestion
```

**Review by erikjohnston — APPROVED** [review]

**Pull request merged**